### ISSUE TITLE
MAML inner-loop `param.data -=` detaches the graph — the meta-model never trains

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#13118]
﻿## Description

`backend/ml/meta/model.py` implements MAML but mutates parameters through `.data` in the inner (adaptation) loop:

```python
# meta/model.py:95,99  inner_update
grads = torch.autograd.grad(loss, adapted_model.parameters(), create_graph=True)
for param, grad in zip(adapted_model.parameters(), grads):
    param.data -= self.inner_lr * grad          # <-- in-place .data assignment

# meta/model.py:116-131  meta_train_step
adapted_model = self.inner_update(self.model, support_x, support_y)
pred = adapted_model(query_x)
task_loss = self.criterion(pred, query_y)
meta_loss += task_loss
...
self.outer_update(meta_loss)                     # meta_loss.backward() on self.model.parameters()
```

The same pattern appears at `meta/model.py:169-172` (`adapt`).

## Expected Behavior

The inner update keeps the computation graph alive so that the query loss `task_loss` (computed through the adapted parameters) back-propagates through the inner step into `self.model.parameters()`. The outer optimizer then actually updates the meta-weights.

## Actual Behavior

Writing through `param.data` **detaches** the inner step from the autograd graph. The query loss graph therefore has no path back to `self.model.parameters()`, so `meta_loss.backward()` leaves their `.grad` as `None` and `outer_update`'s `step()` updates nothing. This is **silent**: no exception, loss numbers print, but the meta-model never learns. (Also occurs in `adapt()` at line 172.)

## Proposed Fix

Use functional updates that preserve the graph (e.g. the `higher` library, or build adapted params as tracked tensors):

```python
grads = torch.autograd.grad(loss, adapted_params, create_graph=True)
adapted_params = [p - self.inner_lr * g for p, g in zip(adapted_params, grads)]
# forward query_x through a functional copy of the network using adapted_params
```

Apply the same change to `adapt()` (line 172). Add a regression test asserting `self.model.parameters()` have non-`None` `.grad` after one `meta_train_step`, and that loss decreases over steps.
